src/odemis/driver/orsay.py

```diff
@@ -1262,3 +1326,1229 @@ def terminate(self):
+        else:
+            self._va_is_tuple = False
+            self._va_value_type = type(self._va.value)
+        if not self._va.readonly:  # only overwrite the VA's setter if the VA is not read-only
+            self._va._setter = WeakMethod(self._update_parameter)
+        if self._va_is_tuple and not len(self._parameters) == len(self._va.value):
+            raise ValueError("Length of Tuple VA does not match number of parameters passed.")
+        if len(self._parameters) > 1 and not self._va_is_tuple:
+            raise ValueError("Multiple parameters are passed, but VA is not of a tuple type.")
+
+        self._factor = None
+        if self._va_value_type == float and factor != 0:
+            self._factor = factor
+
+        # If the VA has a range, check the Orsay server if a range of the parameter is specified and copy this range
+        if hasattr(self._va, "range"):
+            if self._va_is_tuple:
+                new_range = [list(self._va.range[0]), list(self._va.range[1])]
+            else:
+                new_range = [[self._va.range[0]], [self._va.range[1]]]
+
+            for i in range(len(self._parameters)):
+                p = self._parameters[i]
+                # Search for a lowerbound on the server
+                if self._minpar:  # in case a minimum parameter is supplied
+                    if self._minpar[i].Actual is not None:
+                        lowerbound = self._minpar[i].Actual
+                    else:
+                        lowerbound = self._minpar[i].Target
+                    if p.Min is not None and p.Min != lowerbound:
+                        logging.warning("%s.Min and %s contain different, non-None values."
+                                        "Contact Orsay Physics about this!" % (p.Name, self._minpar[i].Name))
+                else:
+                    lowerbound = p.Min
+                if lowerbound is not None:  # if a lowerbound is defined in the server
+                    new_range[0][i] = self._parameter_to_VA_value(lowerbound)  # copy it to the va
+                # Search for an upperbound on the server
+                if self._maxpar:  # in case a minimum parameter is supplied
+                    if self._maxpar[i].Actual is not None:
+                        upperbound = self._maxpar[i].Actual
+                    else:
+                        upperbound = self._maxpar[i].Target
+                    if p.Max is not None and p.Max != upperbound:
+                        logging.warning("%s.Max and %s contain different, non-None values."
+                                        "Contact Orsay Physics about this!" % (p.Name, self._maxpar[i].Name))
+                else:
+                    upperbound = p.Max
+                if upperbound is not None:  # if an upperbound is defined in the server
+                    new_range[1][i] = self._parameter_to_VA_value(upperbound)  # copy it to the va
+
+            if len(new_range[0]) == 1:
```

Comment:
Shouldn't it be `if self._va_is_tuple:`, the same as above. So that it's more explicit.